src/renderers/dom/shared/devtools/ReactDOMUnknownPropertyDevtool.js

```diff
@@ -87,28 +89,57 @@ if (__DEV__) {
       // We were unable to guess which prop the user intended.
       // It is likely that the user was just blindly spreading/forwarding props
       // Components should be careful to only render valid props/attributes.
-      warning(
-        false,
-        'Unknown prop `%s` on <%s> tag. Remove this prop from the element. ' +
-        'For details, see https://fb.me/react-unknown-prop%s',
-        name,
-        tagName,
-        ReactComponentTreeDevtool.getStackAddendumByID(debugID)
-      );
+      unknownProp = name;
     }
+    return unknownProp;
   };
 }
 
+var groupWarnings = function(props, tagName, debugID) {
+  var propString = props.map(function(prop) {
+    return '`' + prop + '`';
+  }).join(', ');
+
+  if (props.length === 1) {
+    warning(
+      false,
+      'Unknown prop %s on <%s> tag. Remove this prop from the element. ' +
+      'For details, see https://fb.me/react-unknown-prop%s',
+      propString,
+      tagName,
+      ReactComponentTreeDevtool.getStackAddendumByID(debugID)
+    );
+  } else if (props.length > 1) {
+    warning(
+      false,
+      'Unknown props %s on <%s> tag. Remove these props from the element. ' +
+      'For details, see https://fb.me/react-unknown-prop%s',
+      propString,
+      tagName,
+      ReactComponentTreeDevtool.getStackAddendumByID(debugID)
+    );
+  }
+};
+
+var warnUnknownProperties = function(debugID, element) {
+  var unknownProps = [];
+  for (var key in element.props) {
+    var prop = handleProperty(element.type, key, debugID);
+    if (prop !== undefined) {
+      unknownProps.push(prop);
+    }
+  }
+  groupWarnings(unknownProps, element.type, debugID);
```

Comment:
Let’s inline `groupWarnings` into this function. It doesn’t do much else and isn’t used anywhere so it gives an impression of being more generic than it actually is.